include/seqan3/alignment/pairwise/edit_distance_unbanded.hpp

```diff
@@ -426,27 +426,28 @@ struct pairwise_alignment_edit_distance_unbanded
     result_type & operator()(result_type & res)
     {
         _compute();
-        if constexpr (std::tuple_size_v<result_type> == 2)
+        if constexpr (std::tuple_size_v<result_type> >= 2)
         {
             get<align_result_key::score>(res) = score();
-        }
-        else if constexpr (std::tuple_size_v<result_type> == 3)
-        {
-            throw std::invalid_argument{"The current output setting is not supported!"};
-        }
-        else if constexpr (std::tuple_size_v<result_type> == 4)
-        {
-            throw std::invalid_argument{"The current output setting is not supported!"};
-        }
-        else if constexpr (std::tuple_size_v<result_type> == 5)
-        {
-            //TODO:
-            auto trace = alignment_trace(database, query, trace_matrix());
-            ranges::copy(std::get<0>(trace), ranges::back_inserter(std::get<0>(get<align_result_key::trace>(res))));
-            ranges::copy(std::get<1>(trace), ranges::back_inserter(std::get<1>(get<align_result_key::trace>(res))));
-            // TODO: Compute the end coordinate
-            // TODO: Compute the begin coordinate
-            get<align_result_key::score>(res) = score();
+
+            if constexpr (std::tuple_size_v<result_type> >= 3)
+            {
+                alignment_coordinate end = end_coordinate();
+                get<align_result_key::end>(res) = end;
+
+                if constexpr (std::tuple_size_v<result_type> >= 4)
```

Comment:
don't nest them

Reply:
If I don't nest it will be a lot of code duplication

Reply:
Why? If condition end is fulfilled we will write the end coordinate.
if condition begin is fulfilled we write begin coordinate.
If we need to store the state then we do it in the class. That is how the rest is done anyway.
if condition trace is fulfilled we write trace. 
This means, that the lower requirements are always included for the higher ones and the order gives us the right way to execute it without duplicating code

Reply:
okay I don't nest it anymore